projects/storefrontlib/src/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -1,67 +1,123 @@
 import {
   ChangeDetectionStrategy,
   Component,
+  HostBinding,
   Input,
-  OnInit,
-  ViewEncapsulation,
+  Optional,
 } from '@angular/core';
 import { FormControl } from '@angular/forms';
-import { merge, Observable, Subject } from 'rxjs';
-import { take } from 'rxjs/operators';
+import { CmsSearchBoxComponent } from '@spartacus/core';
+import { CmsComponentData } from 'projects/storefrontlib/src/cms-structure';
+import { Observable } from 'rxjs';
 import { ICON_TYPES } from '../../../cms-components/misc/icon/index';
 import { SearchBoxComponentService } from './search-box-component.service';
+import { SearchResults } from './search-box.model';
 @Component({
   selector: 'cx-searchbox',
   templateUrl: './search-box.component.html',
-  encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SearchBoxComponent implements OnInit {
-  iconTypes = ICON_TYPES;
+export class SearchBoxComponent {
   searchBoxControl: FormControl = new FormControl();
-  isMobileSearchVisible: boolean;
-
-  queryText$: Subject<string> = new Subject();
 
+  /**
+   * Sets the search box input field
+   */
   @Input('queryText')
   set queryText(value: string) {
-    this.queryText$.next(value);
-    this.searchBoxControl.setValue(value);
+    if (value) {
```

Comment:
if we would like to allow for clearing value with '' using input, it would be better to test `if (value !== undefined)`